src/frontend/src/sol/services/sol-address.services.ts

```diff
@@ -0,0 +1,156 @@
+import {
+	SOLANA_DEVNET_NETWORK_ID,
+	SOLANA_KEY_ID,
+	SOLANA_LOCAL_NETWORK_ID,
+	SOLANA_MAINNET_NETWORK_ID
+} from '$env/networks/networks.sol.env';
+import { SOLANA_TOKEN_ID } from '$env/tokens/tokens.sol.env';
+import {
+	getIdbSolAddressMainnet,
+	setIdbSolAddressDevnet,
+	setIdbSolAddressLocal,
+	setIdbSolAddressMainnet,
+	setIdbSolAddressTestnet,
+	updateIdbSolAddressMainnetLastUsage
+} from '$lib/api/idb.api';
+import { getSchnorrPublicKey } from '$lib/api/signer.api';
+import {
+	certifyAddress,
+	loadIdbTokenAddress,
+	loadTokenAddress,
+	type LoadTokenAddressParams,
+	validateAddress
+} from '$lib/services/address.services';
+import {
+	solAddressDevnetStore,
+	solAddressLocalnetStore,
+	solAddressMainnetStore,
+	solAddressTestnetStore,
+	type StorageAddressData
+} from '$lib/stores/address.store';
+import type { SolAddress } from '$lib/types/address';
+import type { CanisterApiFunctionParams } from '$lib/types/canister';
+import { LoadIdbAddressError } from '$lib/types/errors';
+import type { OptionIdentity } from '$lib/types/identity';
+import type { TokenId } from '$lib/types/token';
+import type { ResultSuccess } from '$lib/types/utils';
+import { getAddressDecoder } from '@solana/addresses';
+import { SOLANA_DERIVATION_PATH_PREFIX } from '$sol/constants/sol.constants';
+import { SolanaNetworks } from '$sol/types/network';
+
+const getSolanaPublicKey = async (
```

Comment:
This method is moved from sol-balance.services.ts as it fits better here. It's already checked with  https://github.com/dfinity/oisy-wallet/pull/3925